libs/api/domains/notifications/src/lib/notifications.model.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  Field,
+  GraphQLISODateTime,
+  ID,
+  InputType,
+  ObjectType,
+  registerEnumType,
+  Int,
+} from '@nestjs/graphql'
+import { PageInfoDto } from '@island.is/nest/pagination'
+import { RenderedNotificationDtoStatusEnum } from '@island.is/clients/user-notification'
+
+registerEnumType(RenderedNotificationDtoStatusEnum, {
+  name: 'NotificationStatus',
+})
+
+@ObjectType('NotificationMetadata')
+export class NotificationMetadata {
+  @Field(() => GraphQLISODateTime)
+  sent!: Date
+
+  @Field(() => GraphQLISODateTime, { nullable: true })
+  updated?: Date
+
+  @Field(() => GraphQLISODateTime, { nullable: true })
+  created?: Date
+
+  @Field(() => GraphQLISODateTime, { nullable: true })
+  read?: Date
+
+  @Field(() => RenderedNotificationDtoStatusEnum)
+  status!: RenderedNotificationDtoStatusEnum
+}
+
+@ObjectType('NotificationSender')
+export class NotificationSender {
+  @Field()
+  name!: string
+
+  @Field({ nullable: true })
+  logo?: string
+}
+
+@ObjectType('NoticiationRecipient')
+export class NotificationRecipient {
+  @Field({ nullable: true })
+  nationalId?: string
+}
+
+@ObjectType('NotificationLink')
+export class NotificationLink {
+  @Field({ nullable: true })
+  uri?: string
+}
+
+@ObjectType('NotificationMessage')
+export class NotificationMessage {
+  @Field()
+  title!: string
+
+  @Field()
+  body!: string
+
+  @Field(() => NotificationLink)
+  link!: NotificationLink
+}
+
+@ObjectType('Notification')
+export class Notification {
+  @Field(() => Int)
+  id!: number
+
+  @Field(() => ID)
+  notificationId!: string
+
+  @Field(() => NotificationMetadata)
+  metadata!: NotificationMetadata
+
+  @Field(() => NotificationSender)
+  sender!: NotificationSender
+
+  @Field(() => NotificationRecipient)
+  recipient!: NotificationRecipient
+
+  @Field(() => NotificationMessage)
+  message!: NotificationMessage
+}
+
+@ObjectType('NotificationMessageCounts')
+export class NotificationMessageCounts {
+  @Field(() => Int, { nullable: true })
+  totalCount?: number
+
+  @Field(() => Int, { nullable: true })
+  unreadCount?: number
+}
+
+@InputType('NotificationsInput')
+export class NotificationsInput {
+  @Field(() => Int, { nullable: true })
+  first?: number
+
+  @Field(() => Int, {
+    nullable: true,
+  })
+  after?: number
+
+  @Field(() => Int, {
+    nullable: true,
+    defaultValue: 10,
+  })
+  limit?: number
+}
+
+@ObjectType('Notifications')
+export class NotificationsResponse {
```

Comment:
There exists a helper function in `@island.is/nest/graphql` to create paginated response types

```suggestion
export class NotificationsResponse extends PaginatedResponse(Notification) { }
```

Reply:
Notification service uses offset based pagination. I was under the impression that the helper function was only for cursor based pagination.

Reply:
I am lying I was looking at the input.